src/LevelLoader.cpp

```diff
@@ -296,3 +296,19 @@ ActionComponent* LevelLoader::getActionComponent(json obj) {
 
    return nullptr;
 }
+
+//TODO(nurgan) find a better method for parsing to enum
+GameObjectType LevelLoader::stringToType(std::string type) {
```

Comment:
Yeah enum <-> string sucks in C++. I think the if-then chain is fine since this will only run on load so I don't think adding the complexity of a map is necessary.

In terms of where to put the logic, leaving it all in `GameObject` makes sense so that the possible types are encapsulated there.